src/components/getting-started/codeblock.js

```diff
@@ -24,115 +24,87 @@ import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
 
-  copyCodeToClipboard(e) {
-    e.preventDefault();
+  const preElement = useRef(null);
 
-    copy(this.promptLinesAsString());
+  const getPromptLinesAsString = () => {
+    const string = React.Children.toArray(children)
+      .filter(x => x.type.displayName === 'Prompt')
+      .map(x => x.props.children)
+      .join('\n');
 
-    this.setState({ clicked: false });
-  }
+    return dedent(string);
+  };
 
-  classNames() {
-    var classNames = [];
+  const copyCodeToClipboard = e => {
+    e.preventDefault();
 
-    // this.props.children is either an array or in the case of 1 child
+    const contentToCopy = getPromptLinesAsString();
+    setClipboardContent(contentToCopy);
+  };
+
+  const getClassNames = () => {
+    // props.children is either an array or in the case of 1 child
     // just that child object
     // So this makes sure I always have an array, and flattens it.
-    // TODO: Use React.Children. It exists to do this kind of stuff for me.
-    var childrenArray = [this.props.children].reduce(function(a, b) {
-      return a.concat(b);
-    }, []);
+    const childrenArray = React.Children.toArray(children);
 
-    classNames.push('codeblock--container');
-    if (this.state.hovering) {
+    const classNames = ['codeblock--container'];
+
+    if (isHovering) {
       classNames.push('hovering');
     }
-    if (this.state.clicked) {
+    if (hasContentInClipboard) {
       classNames.push('clicked');
     }
     if (childrenArray.length === 1) {
       classNames.push('oneline');
     }
 
     return classNames.join(' ');
-  }
-
-  render() {
-    return (
-      <div className={this.classNames()}>
-        <pre>
-          <div
-            className='content'
-            ref={d => {
-              this.pre = d;
-            }}
-          >
-            {this.props.children}
-          </div>
-          <div className='codeblock--buttons'>
-            <a
-              href='#'
-              onClick={this.copyCodeToClipboard.bind(this)}
-              onMouseOut={function() {
-                this.setState({ hovering: false });
-              }.bind(this)}
-              onMouseOver={function() {
-                this.setState({ hovering: true });
-              }.bind(this)}
-              onMouseUp={function() {
-                this.setState({ clicked: true });
-              }.bind(this)}
-            >
-              <i aria-hidden='true' className='fa fa-content-copy' />
-            </a>
-          </div>
-          <ReactCSSTransitionGroup
-            transitionEnterTimeout={1000}
-            transitionLeaveTimeout={1000}
-            transitionName={'checkmark'}
+  };
+
+  return (
+    <div className={getClassNames()}>
+      <pre>
+        <div className='content' ref={preElement}>
+          {children}
+        </div>
+        <div className='codeblock--buttons'>
+          <a
+            href='#'
+            onClick={() => setClipboardContent(null)}
```

Comment:
We're missing now the `e.preventDefault()`, so clicking this causes a bump up to the top of the page.

You can test by creating a cluster and going down to one of the copy buttons on the getting started guide.

Maybe the whole setup here too weird, probably something i arrived at because of the animation flourishes I wanted to add to this action.

Reply:
Oh yes, added that now. Thank you 👍